pkg/arvo/sys/vane/gall.hoon

```diff
@@ -2,1457 +2,2475 @@
+        ap-core
+      ::
+      =.  u.level  (dec u.level)
+      ?:  =(0 u.level)
+        =/  deleted  (~(del by meter.subscribers.current-agent) agent-bone)
+        ap-core(meter.subscribers.current-agent deleted)
+      ::
+      =/  dequeued
+        (~(put by meter.subscribers.current-agent) agent-bone u.level)
+      ap-core(meter.subscribers.current-agent dequeued)
+    ::  +ap-produce-arm: produce arm.
+    ::
+    ++  ap-produce-arm
+      ~/  %ap-produce-arm
+      |=  =term
+      ^-  [(each vase tang) _ap-core]
+      ::
+      =/  compiled
+        =/  =type  p.running-state.current-agent
+        =/  =hoon  [%limb term]
+        (~(mint wa cache.current-agent) type hoon)
+      ::
+      =/  virtual
+        =/  trap  |.(compiled)
+        (mule trap)
+      ::
+      ?:  ?=(%.n -.virtual)
+        =/  =tang  p.virtual
+        [[%.n tang] ap-core]
+      ::
+      =/  possibly-vase=(each vase tang)
+        =/  value  q.running-state.current-agent
+        =/  ton  (mock [value q.+<.virtual] ap-namespace-view)
+        ?-  -.ton
+          %0  [%.y p.+<.virtual p.ton]
+          %1  [%.n (turn p.ton |=(a=* (smyt (path a))))]
+          %2  [%.n p.ton]
+        ==
+      ::
+      =/  next
+        =/  =worm  +>.virtual
+        ap-core(cache.current-agent worm)
+      ::
+      [possibly-vase next]
+    ::  +ap-enqueue: add to queue.
+    ::
+    ::    Every agent has a 'meter', that tracks the number of incoming
+    ::    subscribers by bone.  We get both the meter and ship associated with
+    ::    the current bone; if the meter has hit twenty for another ship, we
+    ::    don't enqueue the subscriber.  Otherwise we increment the meter for
+    ::    the current bone and update the agent's state with it.
     ::
-    ++  ap-fill                                         ::  add to queue
-      ^-  {? _.}
-      =+  suy=(~(gut by qel.ged) ost 0)
+    ::    Returns a yes if the meter has been incremented, and no otherwise.
+    ::
+    ++  ap-enqueue
+      ^-  [? _ap-core]
+      ::
+      =/  meter  (~(gut by meter.subscribers.current-agent) agent-bone 0)
       =/  subscriber=(unit (pair ship path))
-        (~(get by sup.ged) ost)
-      ?:  ?&  =(20 suy)
+        (~(get by incoming.subscribers.current-agent) agent-bone)
+      ::
+      ?:  ?&  =(20 meter)
               ?|  ?=(~ subscriber)
                   !=(our p.u.subscriber)
               ==
           ==
-        ~&  [%gall-pulling-20 ost (~(get by sup.ged) ost) (~(get by r.zam) ost)]
-        [%| ..ap-fill]
-      ::  ~&  :*  %gall-pushing-20
-      ::          ost
-      ::          suy=suy
-      ::          (~(get by r.zam) ost)
-      ::      ==
-      [%& ..ap-fill(qel.ged (~(put by qel.ged) ost +(suy)))]
-    ::
-    ++  ap-find                                         ::  general arm
-      ~/  %ap-find
-      |=  {cog/term pax/path}
-      ^-  [(unit (pair @ud term)) _+>]
-      ::  check cache
-      ?^  maybe-result=(~(get by arms) [cog pax])
```

Comment:
My previous question here still goes unanswered. Would tracking/providing these be fairly straightforward for you to implement at this point? If not, can we get a TODO marker on this?

Reply:
Yeah that's better. Thanks!